firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -499,6 +499,10 @@ public Target encodeTarget(TargetData targetData) {
       builder.setResumeToken(targetData.getResumeToken());
     }
 
+    if (targetData.getExpectedCount() != null) {
+      builder.setExpectedCount(Int32Value.newBuilder().setValue(targetData.getExpectedCount()));
```

Comment:
This code block is inconsistent with the web sdk. In the web sdk, it only sets expectedCount if a resume token or snapshot version is specified. Here, it is setting expectedCount regardless of whether or not a resume token or snapshot version is specified.

We generally aim to keep the sdks in sync as much as possible because it's confusing when things are done differently in one sdk compared to another.

Please make the two SDKs do the same thing in this scenario.

I think the way that the web sdk does it is best because it makes it very clear that expectedCount is only meaningful if a resume token or snapshot version is used.

https://github.com/firebase/firebase-js-sdk/blob/dd6683583b7c6b07586c2551f4d362b48fb83f7c/packages/firestore/src/remote/serializer.ts#L1013-L1025

Reply:
yeah, this is the part that I am a bit hesitated. In web the if statments is like: 
```
if (targetData.resumeToken.approximateByteSize() > 0) {
  } else if (targetData.snapshotVersion.compareTo(SnapshotVersion.min()) > 0) {
```
while in Android it is 
```
if (targetData.getResumeToken().isEmpty()
        && targetData.getSnapshotVersion().compareTo(SnapshotVersion.NONE) > 0) {
} else{
}
```

Reply:
updated the if statement to indicate resume token or snapshotVersion is required to add expectedCount: 
```
   if (targetData.getExpectedCount() != null
        && (!targetData.getResumeToken().isEmpty()
            || targetData.getSnapshotVersion().compareTo(SnapshotVersion.NONE) > 0)) {
      builder.setExpectedCount(Int32Value.newBuilder().setValue(targetData.getExpectedCount()));
    }
```

Reply:
Even though the "if" statement has the extra check for `targetData.getResumeToken().isEmpty()`, we should still mirror the structure from the web sdk. We only want to set expected count if either the "if" or the "else" branch above is entered, so we may as well put the logic into those branches.